mongo/filterstore.go

```diff
@@ -126,12 +133,15 @@ func (s *FilterStore) AddFilterDimension(filterID, name string, options []string
 		list = append(list, d)
 	}
 
-	queryFilter := bson.M{"filter_id": filterID}
-	update := bson.M{"$set": bson.M{"dimensions": list}}
+	queryFilter := bson.M{"filter_id": filterID, "unique_timestamp": timestamp}
```

Comment:
`queryFilter` would be better off being called `selectFilter` or even better to be just `select`; the Update query includes select and update parts

This can be applied to all update queries in filterstore